client-telemetry-reporter-plugin/kafka/README.md

```diff
@@ -0,0 +1,209 @@
+to monitor Kafka clients.
+
+This tutorial adds `ClientOtlpMetricsReporter` that aggregates [OpenTelemetry Protocol](https://opentelemetry.io/docs/specs/otel/protocol/) (OTLP) metrics
+received from client, enhances them with additional client labels and forwards them via gRPC client
+to an external OTLP receiver. The OTLP receiver can be any OpenTelemetry compatible backend like
+Prometheus, Jaeger, Zipkin, etc. The plugin is implemented in Java and requires the jar to be
+added to the Kafka broker libs.
+
+The OTLP receiver for the tutorial is [otel-collector](https://opentelemetry.io/docs/collector/), which is a part of the OpenTelemetry project.
+The `otel-collector` receives the metrics from the Kafka broker and forwards them to a Prometheus
+instance for visualization.
+
+## Prerequisites
+
+- Java 17 or higher
+- [Otel Collector](#otel-collector)
+- [Prometheus](#prometheus)
+- [Client Telemetry Reporter Plugin jar](#build-the-client-telemetry-reporter-plugin)
+- [Apache Kafka Broker 3.7.0 or higher](#apache-kafka-broker)
+- [Apache Kafka Client 3.7.0 or higher](#create-a-kafka-client)
+
+### Otel Collector
+
+Pull the otel-collector docker image:
+
+```shell
+docker pull otel/opentelemetry-collector-contrib
+```
+
+#### Configuration
+
+Create a configuration file `config.yaml` for the otel-collector.
+
+Example: Below configuration defines a pipeline that receives OTLP metrics from Kafka broker on `4317 gRPC port`
+and forwards them to `Prometheus on 8889 port`. The configuration adds a namespace `kip-714` to the metrics
+which can be used to identify the source of the metrics. The configuration also enables resource to telemetry
+conversion to add labels to the metrics.
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: 0.0.0.0:4317
+
+processors:
+  batch:
+    send_batch_max_size: 100
+    send_batch_size: 10
+    timeout: 10s
+
+exporters:
+  prometheus:
+    endpoint: 0.0.0.0:8889
+    namespace: kip-714
+    resource_to_telemetry_conversion:
+      enabled: true
+
+service:
+  pipelines:
+    metrics:
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [prometheus]
+
+```
+
+#### Start the otel-collector
+
+Replace `<local_path>` with the path to the `config.yaml` file.
+
+```shell
+docker run \
+    -p 127.0.0.1:4317:4317 -p 8889:8889 -p 8888:8888 \
+    -v <local_path>/config.yaml:/etc/otelcol-contrib/config.yaml \
+    otel/opentelemetry-collector-contrib
+ ```
+
+### Prometheus
+
+Download prometheus from [here](https://prometheus.io/download/).
+
+#### Configuration
+
+Update configuration file for prometheus `prometheus.yml` to scrape metrics from otel-collector.
+```yaml
+global:
+  scrape_interval: 15s
+  evaluation_interval: 15s
+
+scrape_configs:
+  - job_name: "prometheus"
+    static_configs:
+      - targets: ["localhost:8889"]
+```
+
+#### Start Prometheus
+
+```shell
+./prometheus --config.file=prometheus.yml
+```
```

Comment:
I think that this block should be redone to use Docker Compose and the repo should include the config files rather than having readers copy/paste. Will be much easier for readers, plus cleanup gets simpler. I ended up doing this because I wasn't able to run the downloaded prometheus binary on my work laptop so I tried to use Docker instead, but then in order for the OTEL collector and prometheus to communicate it's easier to use Docker compose rather than `docker run` commands. Here are the steps

1. add `otel-collector-config.yaml` to the repo (in `client-telemetry-reporter-plugin/kafka`) with same contents as in this PR
2. add `prometheus.yml` to the repo but change it to target the otel container name rather than localhost:
```
global:
  scrape_interval: 15s
  evaluation_interval: 15s

scrape_configs:
  - job_name: "prometheus"
    static_configs:
      - targets: ["otel-collector:8889"]
```

Add `docker-compose.yml` file to the repo that runs the collector and prometheus using those configs:

```
version: '3'
services:
  otel-collector:
    image: otel/opentelemetry-collector-contrib
    container_name: otel-collector
    hostname: otel-collector
    ports:
      - "4317:4317"
      - "8888:8888"
      - "8889:8889"
    volumes:
      - ./otel-collector-config.yaml:/etc/otelcol-contrib/config.yaml

  prometheus:
    image: prom/prometheus
    container_name: prometheus
    hostname: prometheus
    ports:
      - "9090:9090"
    volumes:
      - ./prometheus.yml:/etc/prometheus/prometheus.yml
    depends_on:
      - otel-collector
```

Then trim the README way down with something like the following:

### Start OpenTelemetry Collector and Prometheus

Clone the `confluentinc/tutorials` GitHub repository (if you haven't already) and navigate to the tutorials directory:

```shell
git clone git@github.com:confluentinc/tutorials.git
cd tutorials
```

Start the OpenTelemetry collector and Prometheus:

```shell
docker compose -f ./client-telemetry-reporter-plugin/kafka/docker-compose.yml up -d
```

In this example, we've configured the OTel collector (in `client-telemetry-reporter-plugin/kafka/otel-collector-config.yaml`) to receive OTLP metrics from the Kafka broker via gRPC on port 4317
and forwards them to Prometheus on port 8889. The configuration adds a namespace `kip-714` to the metrics which can be used to identify the source of the metrics. The configuration also enables resource to telemetry conversion to add labels to the metrics.

Reply:
This is very helpful, thanks a lot. I have added this and verified by running the commands.